### PR TITLE
fix: Use a context manager for opening files

### LINKED ISSUE #108: `test_pytket_with_wasm` fails when run after `test_pytket_to_phir_no_machine_at_all`
These 2 tests are probably interacting somehow. When `test_pytket_with_wasm` runs first, it passes, but when `test_pytket_phir_no_machine_at_all` runs first, `test_pytket_with_wasm` fails. It is weird that moving these tests around within the file changes whether they pass or not, they should be totally independent.

This also occurs on the 3q-rz-bug branch when run after `test_three_qubit_rz_exec_order_preserved`.

### PR DESCRIPTION
**Pull request by qartik**
# Description

`ruff` [pointed out](https://github.com/CQCL/pytket-phir/actions/runs/10517561194/job/29141900659) errors like the following:
```
pytket/phir/api.py:94:25: SIM115 Use a context manager for opening files
   |
92 |     if wasm_bytes:
93 |         try:
94 |             qasm_file = NamedTemporaryFile(suffix=".qasm", delete=False)
   |                         ^^^^^^^^^^^^^^^^^^ SIM115
95 |             wasm_file = NamedTemporaryFile(suffix=".wasm", delete=False)
96 |             qasm_file.write(qasm.encode())
   |
```

Turns out resolving these also fixes #108.

Additionally:
- fixes warning "DeprecationWarning: The `auto_rebase_pass()` method is deprecated and will be removed after pytket 1.33, please use `AutoRebase` instead."
- update deps

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [x] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog with any user-facing changes

### CODE REVIEW

File: pytket/phir/api.py [hunk 1]
```diff
@@ -90,22 +89,18 @@ def qasm_to_phir(
     """
     circuit: Circuit
     if wasm_bytes:
-        try:
-            qasm_file = NamedTemporaryFile(suffix=".qasm", delete=False)
-            wasm_file = NamedTemporaryFile(suffix=".wasm", delete=False)
+        with (
+            NamedTemporaryFile(suffix=".qasm", delete=False) as qasm_file,
+            NamedTemporaryFile(suffix=".wasm", delete=False) as wasm_file,
```
Comment (qartik) [inline pytket/phir/api.py:94]: It's not obvious to me if we need `delete=False` anymore.

**Review by cqc-alec — APPROVED** [review]

File: pytket/phir/api.py (hunk 1 above)
Comment (qartik) [reply pytket/phir/api.py:94]: Perhaps @neal-erickson might have insight about why `delete=False` was needed?

**Comment by qartik** [comment]
Hi @peter-campora, would it possible for you to try this branch on Windows?

@neal-erickson shared:
> the reason I wasn't using a context manager was because it seemed to be broken on Windows. I don't remember the exact details, I think I had Pete test something for me. But this commit where I changed it from a context manager to the try/finally makes me think there's a good reason.
>
> [`f2e72af` (#77)](https://github.com/CQCL/pytket-phir/pull/77/commits/f2e72aff73c814c74c4710985a312d52cd38729b)
>
> But maybe you can work with him to make sure things are working on Windows?

I do see the CI on windows is passing fully, so I am unsure if there is any problem still unfixed on Windows.

**Comment by peter-campora** [comment]
I forget what the status of python tooling natively in my windows environment is exactly, but I can take a look if still desired that I do some testing

**Comment by qartik** [comment]
> I forget what the status of python tooling natively in my windows environment is exactly, but I can take a look if still desired that I do some testing

If you don't mind, it will be helpful to eliminate the possibility that this is still an issue.

**Comment by qartik** [comment]
Going to merge this. If it still fails on Windows, we will fix it before the next release.

**Pull request merged**